templates/tfengine/components/devops/variables.tf

```diff
@@ -50,32 +54,45 @@ variable "parent_id" {
   }
 }
 
+variable "parent_type" {
+  type        = string
+  description = {{schemaDescription $props.parent_type.description}}
+  validation {
+    condition     = can(regex("{{$props.parent_type.pattern}}", var.parent_type))
+    error_message = "The parent_type must be valid. Should have only numeric values with a length between 8 and 25 digits. See https://cloud.google.com/resource-manager/docs/cloud-platform-resource-hierarchy to know how to get your organization/folder id."
+  }
+}
+
 variable "project" {
   type = object({
     apis = list(string)
     owners_group = object({
-      id           = string
-      {{- $missing_project_owners_group := not (get .project.owners_group "exists")}}
-      {{- if $missing_project_owners_group}}
       customer_id  = string
-      display_name = string
-      {{- if has .project.owners_group "description"}}
       description  = string
-      {{- end}}
-      {{- if has .project.owners_group "owners"}}
+      display_name = string
+      exists       = string
+      id           = string
       owners       = list(string)
-      {{- end}}
-      {{- if has .project.owners_group "managers"}}
       managers     = list(string)
-      {{- end}}
-      {{- if has .project.owners_group "members"}}
       members      = list(string)
-      {{- end}}
-      {{- end}}
     })
     project_id = string
   })
-  description = {{schemaDescription $props.project.description}}
+  description = <<EOF
+    {{$props.project.description}}
+
+    Fields:
+
+    * apis = {{$projectProps.apis.description}}
+    * owners_group = {{$projectProps.owners_group.description}}
+    ** customer_id {{$projectOwnersGroupProps.customer_id.description}}
+    ** description {{$projectOwnersGroupProps.description.description}}
+    ** display_name {{$projectOwnersGroupProps.display_name.description}}
+    ** exists {{$projectOwnersGroupProps.exists.description}}
```

Comment:
Can we use extra indent instead of double *?

Reply:
We can, but that might add extra complexity if we automate this by iterating the properties as Martin suggested in other PRs.

What do you think might be the best?

Reply:
Maybe this should be part of the schemaDescription function, to also output the internal field descriptions. Then you could iterate and use recursion (for nested objects) with normal code instead of templating code. Would that be complex to do?

Reply:
Sounds doable and better, but the problems are almost the same:

1. Sometimes properties are inside properties key or items key (this is fixable if I understand better when it's one or the other)
2. Some schema variables may be renamed into the `variables.tf` or don't appear at all. How can we detect those?

Reply:
Good point.

Then in the interest of not debating this forever, I'm okay with not iterating for now. Besides that, I agree that an extra indent would look better than 2 **s

Reply:
Got it. Looks better, thanks!